### DISCUSSION TITLE
Welcome to Odin Discussions!

### DISCUSSION THREAD

**Discussion by gingerBill** [odin-lang/Odin#803] [General]
## 👋 Hello and Welcome!
Odin is now using GitHub's Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about regarding Odin
  * Share ideas for the language, core library, tooling, documentation, and more!
  * Engage with other community members
  * Help others in need
  * Show off your latest projects in Odin

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ghost** [comment]
Yay! You got Github Discussions!

I'm Christian "krixano" Seibold. I program in Odin, working on some projects - Orpheus, Paled, and a few others.

**Comment by Tetralux** [comment]
"And so it begins." 😁

Since getting more involved with Odin about a year ago, Odin's become one of my most-used languages, and has been my go-to for most low-level programming I do.

I tend to be fairly active in the Discord, hanging out and having fun with people, and help to answer people's questions.

**Comment by flysand7** [comment]
My internet name is `bumbread` (I never tell a real one to anyone on the internet `:(` ). I'm a student currently learning programming and various things about computers in general. So far I have spent about 3 hours total with Odin and I can say that I enjoy it almost as much as I enjoyed C (I still have to learn about things and Odin has yet to get more powerful too, soo..)

...

I'm assuming this one is a great place to be asking things that aren't part of any Odin documentation and examples. The things that you can't learn just by looking at them. I'm talking about the decisions, and the philosophy behind developing Odin.

Recently I have shown Odin to someone, and the first reaction was "This is like rust.. oh wait, what's this, `defer` statement? looks like they're playing with fire". So I'm going to start the discussion by asking, how Odin differs from Rust, and what could be the reasons to prefer one language over another?

**Comment by ThisDevDane** [comment]
I am the CI plumber and official "Nag Bill about everything"-guy.

**Comment by vassvik** [comment]
I made a few popular C bindings and wrappers:
 - https://github.com/vassvik/odin-glfw (both a binding and wrapper, with some convenience stuff added) 
 - https://github.com/vassvik/odin-gl (a port of glad 4.5 core) 
 - https://github.com/vassvik/odin-stb (bindings for some of the popular single header libraries ny Sean Barrett) 
 - https://github.com/vassvik/odin-gl_font (a basic and rough font rendering library using the preceeding three libraries)

In addition I work at JangaFX, where we use Odin in production for our main product, EmberGen, a real-time fluid simulation software for creating visual effects for film and games. We currently employ 9 people, and it's been an interesting experience to see how people who are not immediately familiar with Odin are able to adapt to a new language.

**Comment by hasenj** [comment]
I'm a web developer by day and use odin for my side projects (although I haven't been having much free time to do that lately).

Although it's not my most used language (yet) it's definitely my favorite language, and the language I plan to use going forward.

I hope I can contribute something useful in terms of libraries

**Comment by oskarnp** [comment]
Hello. I've been programming in C and C++ since the early 2000s and professionally using C since 2007 for general systems programming and embedded systems. I've also done freelance contracting as a C++ engine programmer for a PC video game.

I started writing my own toy game engine in C++ several years ago, but like many others got annoyed with C++ so started researching newer languages like Rust, Zig, and Odin. I eventually settled on Odin.

I am pretty active in the Discord server and enjoys helping and answering questions as much as I can.

**Comment by HardCoreCodin** [comment]
Started looking into alternatives to C/C++ earlier this year (Zig, Jai, Beef, etc.) and landed on Odin. 
Ported my software raytracer from C to Odin and started a [YouTube playlist](https://www.youtube.com/playlist?list=PLKkGMhjiYwWdAub0WzbTMAjHRnG-Egucx) showing how to write one in Odin from scratch.
Adapted it into a [Wolfestein3D style 2D raycaster](https://youtu.be/RxBg8BsLVxo), adding texture filtering, lighting and an editable minimap.
This project recently got featured in javidx9's [One Lone Coder Community Showcase 2020](https://youtu.be/Qmr4BW0FK2s?t=1179)
Semi-active on discord, though recently had mostly focused on a more feature rich software raytracer in C with optional CUDA support.
Plan to re-visit my Odin playlist soon'ish, providing additional Odin codestreams focusing on graphics programming :)

**Comment by PrivacyDragon** [comment]
I'm fairly new to programming and started with 'seriously' programming in 2020. I mostly make calculator programs for the TI-84+ CE. On the calculator programming forum was a happy holidays coding challenge, where the goal was to make a Christmas tree in as much languages as possible. (https://ceme.tech/p289742)
That was when I found out about the Odin language.  And the name made me interested.  (I like the old Norse gods and the Edda's)
